### PR TITLE
refactor: filemanager stack

### LINKED ISSUE #169: filemanager: reconfigure deployment to match other microservices
The deployment should be reconfigured to match other microservices. This includes:

* Change filemanager from construct to stack.
* Check cdk-nag tests to ensure filemanager passes/ignores relevant lints using the `microserviceStackArray` pattern.
* Make use of the postgres manager - related to #138.

### PR DESCRIPTION
**Pull request by mmalenic**
Closes #169 

### Changes
* As described in #169 except using `RDS_IAM` as part of postgres manager. Note, currently this still uses the master RDS credentials. Both these points will be addressed in #138.
* Move `CdkResourceInvoke` to `components`.
* Add stg and prod filemanager deployments.

### CODE REVIEW

File: test/stateless/stateless-deployment.test.ts [hunk 1]
```diff
@@ -99,6 +99,17 @@ function applyNagSuppression(stackId: string, stack: Stack) {
...
+    stack,
+    [
+      {
+        id: 'AwsSolutions-L1',
+        reason: "'AwsCustomResource' is out of date",
+      },
+    ],
+    true
+  );
+
```
Comment (mmalenic) [inline test/stateless/stateless-deployment.test.ts:112]: This will probably change with #170.

File: lib/workload/orcabus-stateless-stack.ts [hunk 2]
```diff
@@ -70,13 +55,7 @@ export class OrcaBusStatelessStack extends cdk.Stack {
...
     this.microserviceStackArray.push(this.createSequenceRunManager(props));
     this.microserviceStackArray.push(this.createPostgresManager(props.postgresManagerConfig));
-
-    if (props.filemanagerDependencies) {
-      this.createFilemanager({
-        ...props.filemanagerDependencies,
-        lambdaSecurityGroupName: props.lambdaSecurityGroupName,
-      });
-    }
+    this.microserviceStackArray.push(this.createFilemanager(props.filemanagerConfig));
```
Comment (victorskl) [inline lib/workload/orcabus-stateless-stack.ts:58]: Woo ho ho! Welcome to the nag club! 🤭

File: lib/workload/stateful/event_source/component.ts [hunk 3]
```diff
@@ -55,9 +55,10 @@ export class EventSource extends Construct {
   constructor(scope: Construct, id: string, props: EventSourceProps) {
     super(scope, id);
 
-    this.deadLetterQueue = new Queue(this, 'DeadLetterQueue');
+    this.deadLetterQueue = new Queue(this, 'DeadLetterQueue', { enforceSSL: true });
     this.queue = new Queue(this, 'Queue', {
       queueName: props.queueName,
+      enforceSSL: true,
```
Comment (victorskl) [inline lib/workload/stateful/event_source/component.ts:61]: Nice!

**Review by victorskl — APPROVED** [review]
LGTM.!

(could you hold merge this a bit, tick, pls. let me fix the SRM unit test PR real quick.)

**Review by williamputraintan — APPROVED** [review]

**Pull request merged**